googleapis_auth/lib/src/oauth2_flows/token_model.dart

```diff
@@ -0,0 +1,174 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:js';
+
+import '../access_credentials.dart';
+import '../access_token.dart';
+import '../authentication_exception.dart';
+import '../browser_utils.dart';
+import '../utils.dart';
+import 'token_model_interop.dart' as interop;
+
+JsObject get _googleAccountsId =>
+    ((context['google'] as JsObject)['accounts'] as JsObject)['id'] as JsObject;
+
+/// Obtains [AccessCredentials] using the
+/// [Google Identity Services](https://developers.google.com/identity/oauth2/web/guides/overview)
+/// token model.
+///
+/// See
+/// [Choose a user authorization model](https://developers.google.com/identity/oauth2/web/guides/choose-authorization-model)
+/// to understand the tradeoffs between using this function and
+/// [requestAuthorizationCode].
+///
+/// See https://developers.google.com/identity/oauth2/web/guides/use-token-model
+/// and https://developers.google.com/identity/oauth2/web/reference/js-reference
+/// for more details.
+Future<AccessCredentials> requestAccessCredentials({
+  required String clientId,
+  required Iterable<String> scopes,
+  String prompt = 'select_account',
+  @Deprecated('Undocumented feature. Do not include in production code.')
+      String? logLevel,
+}) async {
+  await initializeScript('https://accounts.google.com/gsi/client');
+  if (logLevel != null) _googleAccountsId.callMethod('setLogLevel', [logLevel]);
```

Comment:
Can you check `if (debug && logLevel != null)`?

Reply:
Which `debug`? Where does `debug` come from?

Reply:
Nevermind, I was thinking of Flutter's `kDebugMode` :)